### PR TITLE
Midi music

### PR DESCRIPTION
**Pull request by stonedDiscord**
i haven't found any issues with this

### CODE REVIEW

**Comment by stonedDiscord** [comment]
actually there is 1 issue, the CI failed for some reason

**Comment by oldmud0** [comment]
Where does it load a soundfont from? Does it sound remotely good?

**Comment by Crystalwarrior** [comment]
what about .mod, .xm, .it etc.

**Comment by stonedDiscord** [comment]
> Where does it load a soundfont from?
same filename but with soundfont ending, is handled by bass
> Does it sound remotely good?
depends on the midi+soundfont really, it ranges from poop midi to better than DS for AA music
> what about .mod, .xm, .it etc.
forgot about those, will add

**Comment by Crystalwarrior** [comment]
> > Where does it load a soundfont from?
> 
> same filename but with soundfont ending, is handled by bass
> 
> > Does it sound remotely good?
> 
> depends on the midi+soundfont really, it ranges from poop midi to better than DS for AA music
> 
> > what about .mod, .xm, .it etc.
> 
> bass already supports those without a plugin

If bass supports it ao doesn't atm

**Comment by stonedDiscord** [comment]
tested and working with echoing.mod and milky.xm

**Comment by Crystalwarrior** [comment]
Missing file checker tests if the file exists using the file path for the music file, it's probably wacky in some silly way

File: src/text_file_functions.cpp [hunk 1]
```diff
@@ -512,7 +512,7 @@ QString AOApplication::get_court_sfx(QString p_identifier, QString p_misc)
 QString AOApplication::get_sfx_suffix(VPath sound_to_check)
 {
   return get_real_suffixed_path(sound_to_check,
-                                {".opus", ".ogg", ".mp3", ".wav" });
+                                {".opus", ".ogg", ".mp3", ".wav", ".midi", ".xm", ".it", ".s3m", ".mod", ".mtm", ".umx" });
```
Comment (oldmud0) [inline src/text_file_functions.cpp:518]: can you add `.mid`

File: src/aomusicplayer.cpp [hunk 2]
```diff
@@ -34,12 +34,18 @@ QString AOMusicPlayer::play(QString p_song, int channel, bool loop,
...
+    else if (f_path.endsWith(".mid"))
+      newstream = BASS_MIDI_StreamCreateURL(f_path.toStdString().c_str(), 0, streaming_flags, nullptr, 0, 1);
     else
       newstream = BASS_StreamCreateURL(f_path.toStdString().c_str(), 0, streaming_flags, nullptr, 0);
 
   } else {
     if (f_path.endsWith(".opus"))
       newstream = BASS_OPUS_StreamCreateFile(FALSE, f_path.utf16(), 0, 0, flags);
+    else if (f_path.endsWith(".mid"))
+      newstream = BASS_MIDI_StreamCreateFile(FALSE, f_path.utf16(), 0, 0, flags, 1);
```
Comment (oldmud0) [inline src/aomusicplayer.cpp:46]: How to set soundfont??

Comment (stonedDiscord) [reply src/aomusicplayer.cpp:46]: BASS doesn't expose this and will use the same filename but with sf2 as file extenstion

**Review by Crystalwarrior — APPROVED** [review]

**Pull request merged**